gr-uhd/lib/usrp_source_impl.cc

```diff
@@ -659,12 +660,43 @@ namespace gr {
 
       case ::uhd::rx_metadata_t::ERROR_CODE_OVERFLOW:
         _tag_now = true;
+
+        //publish message, only if someone's listening
+        if(! pmt::is_null(d_message_subscribers) )
+        {
+          pmt::pmt_t dic = pmt::make_dict();
+          dic = pmt::dict_add(dic, pmt::mp("code"), pmt::from_long(_metadata.error_code));
```

Comment:
I was trying to avoid the case where the message we send is a dictionary with a single entry "code", where the code is "no error".

I'm still not entirely convinced about this anyway. If you're running at multiple 100 Msps, you will get a lot of work calls, and a lot of consequent messages.

If this was only on:
- Errors (O, U, S, D)
- Re-tunes
- Any time an rx_time gets written
- Or even a selectable subset of these

That would seem more manageable.